tests/components/template/test_sensor.py

```diff
@@ -438,6 +442,36 @@ def test_available_template_with_entities(self):
         )
 
 
+async def test_creating_sensor_loads_group(hass):
+    """Test setting up template sensor loads group component first."""
+    order = []
+
+    def register_setup(domain):
+        async def async_setup(hass, config):
+            if domain == "group":
+                # Make sure group takes longer to load, so that it won't
+                # be loaded first by chance
+                await sleep(0.1)
+
+            order.append(domain)
+            return True
+
+        return async_setup
+
+    with patch(
+        "homeassistant.components.group.async_setup", new=register_setup("group"),
+    ):
+        with patch(
+            "homeassistant.components.sensor.async_setup", new=register_setup("sensor"),
+        ):
+            await async_from_config_dict(
+                {"sensor": {"platform": "template", "sensors": {}}, "group": {}}, hass
+            )
+            await hass.async_block_till_done()
+
+    assert ["group", "sensor"] == order
```

Comment:
I see now why this fails. `sensor` is setting up `template` here, and that's done outside of the normal "setup flow" in bootstrap which respects `after_dependencies`.

Reply:
This is the method that doesn't respect it https://github.com/home-assistant/core/blob/dev/homeassistant/setup.py#L235